### DISCUSSION TITLE
Animation of mechanical assembly

### DISCUSSION THREAD

**Discussion by macdroid53** [JacquesLucke/animation_nodes#1853] [Q&A]
I'm thinking there is a more elegant and concise way to animate the disassembly/assembly of mechanical assemblies than what I have cobbled together here. Can anyone share some pointers please? I've looked at some YouTube videos, but they seem to be all oriented to spacial effects or I can't translate the concepts they show to this application.
![Peek 2022-04-29 14-34](https://user-images.githubusercontent.com/6831752/166004885-67f3731c-68da-46b0-b073-074b0cd1e2c7.gif)

[Simple_assy_007.zip](https://github.com/JacquesLucke/animation_nodes/files/8592953/Simple_assy_007.zip)

**Comment by OmarEmaraDev** [comment]
You can use the Delay Falloff node here with manually specified offsets. Add an Index ID Key to the objects you want to animate and add them to a list. The ID key value should be an animation offset, meaning that an object with value 0 will move before an object with value 1. Here I assigned 0, 1, 2, 3 to the bolts and 6 to both other objects in order for them to move together a bit later.

https://user-images.githubusercontent.com/25300994/166312028-c259e077-adad-4200-95d9-a1c233ec1c79.mp4

**Reply by macdroid53** [reply]
I'm looking at your example. I suspected it would involve lists and falloffs, I jut couldn't make the examples I found fit what I wanted to do. I am attempting to duplicate what you show. I don't have a lot of depth in Blender. And entering "blender set object index id key" in a search engine results in a mind numbing list that doesn't help me figure out just how to do that. (This is probably dead easy and obvious, but this is how Swiss cheese my level of Blender foo is...sorry.) Could you point some specific doc or example of adding keys?

**Reply by OmarEmaraDev** [reply]
@macdroid53  You can add that from the AN panel in the N menu.

https://user-images.githubusercontent.com/25300994/166474879-0bfce203-6695-467d-858f-a4ac3905b1b5.mp4

**Reply by macdroid53** [reply]
Got it. Thanks!

**Comment by macdroid53** [comment]
I see how the fall off, etc. node accomplish the movement, and have figured what is going on here for the most part.
Now I'm trying to think through, I guess I'll call sequencing. For instance, once the screws and yokes are out of the way, the gear needs to move off the shaft, then the shaft goes up.
I can put an index on the shaft so that i delays enough, but, the gear isn't that simple, how do I get it to move before the the shaft.
Then get the gear to collapse back onto the shaft...do I make a parallel node sequence?  
I guess I don't know enough to think this through...

**Reply by OmarEmaraDev** [reply]
It would be good if you can illustrate this somehow so that I get a good idea of what you are trying to do.

**Reply by macdroid53** [reply]
> It would be good if you can illustrate this somehow so that I get a good idea of what you are trying to do.

Like as shown. But, the video was created in a rather hzm handed way, and part of the question is how to do long sequential animations in a more concise way.

**Reply by OmarEmaraDev** [reply]
@macdroid53 You will probably have to manually sequence those animations, as I can't think of a good way to make them in a non-overly complicated way.

**Reply by macdroid53** [reply]
That's ok, I just wanted see if there was better ways I need to learn.

But, given that, I have not figured out exactly how to do long sequences.

Do I just chain them together as a long list of nodes? Seems this could get unwieldy.  Is there a way to trigger another chain of nodes to activate after one has completed? 

I have not experimented with multiple node trees.

**Reply by OmarEmaraDev** [reply]
Sorry about the delay. I don't have much time at the moment, so I will get back to you when I do.

**Reply by macdroid53** [reply]
No problem.  I've been experimenting and learning. Still have plenty to learn and will look forward to any info.

**Comment by macdroid53** [comment]
Like this
![Peek 2022-04-29 18-32](https://user-images.githubusercontent.com/6831752/166802912-1b01c9e6-963d-49b8-8736-fd7b97c6ad8f.gif)
.
